File: example/notify/test_notify.js
const baihu = require('baihu');

/**
 * 内建通知测试示例 (Node.js)
 * 
 * 使用说明：
 * 1. 确保已在该 Node.js 环境下安装过内建包。
 * 2. 系统会自动注入 BHPKG_NOTIFY_TOKEN 和 BHPKG_NOTIFY_CHANNEL。
 */

console.log("正在尝试发送 Node.js 内建通知...");

try {
    // 简单的一行代码即可完成推送
    baihu.notify(
        "Node.js 任务提醒", 
        "这是一条来自 Node.js 示例脚本的通知消息。无需配置 API 地址或 Token。"
    );
    
    console.log("发送请求已提交。");
    console.log("提示：内建包采用异步非阻塞发送，不会干扰主逻辑执行。");
    
} catch (e) {
    console.error(`通知失败: ${e.message}`);
    console.error("请检查环境变量是否注入，或是否已运行过 'baihu builtininstall'。");
}

File: example/notify/test_notify.py
import baihu

# 内建通知测试示例 (Python)
# 
# 前提条件：
# 1. 已经在环境中安装了 baihu 包（例如通过 `baihu builtininstall`）
# 2. 环境中已注入有效环境变量：
#    - BHPKG_NOTIFY_TOKEN
#    - BHPKG_NOTIFY_CHANNEL

def main():
    print("正在尝试发送 Python 内建通知...")
    try:
        # 调用内建 notify 函数
        # 内部会自动使用 BHPKG_NOTIFY_TOKEN & BHPKG_NOTIFY_CHANNEL 进行鉴权和投递
        response = baihu.notify(
            title="Python 任务提醒",
            text="这是一条来自 Python 示例脚本的通知消息。调用非常简单！"
        )
        print("发送请求已处理。")
        if response:
            print(f"服务器响应: {response}")
            
    except ImportError as e:
        print(f"错误: 库未正确加载。请确保已执行过环境初始化。详情: {e}")
    except Exception as e:
        print(f"发送过程发生异常: {e}")

if __name__ == "__main__":
    main()
